presto-raptor/src/main/java/com/facebook/presto/raptor/storage/ShardRecoveryManager.java

```diff
@@ -128,9 +128,12 @@ private void enqueueMissingShards()
 
     private Set<UUID> getMissingShards()
     {
-        Set<UUID> shardUuids = shardManager.getNodeShards(nodeIdentifier);
         ImmutableSet.Builder<UUID> missingShards = ImmutableSet.builder();
-        shardUuids.stream().filter(shardUuid -> !storageService.getStorageFile(shardUuid).exists()).forEach(missingShards::add);
+        shardManager.getNodeShards(nodeIdentifier).stream().filter(shardUuid -> {
+            File storageFile = storageService.getStorageFile(shardUuid);
+            File backupFile = storageService.getBackupFile(shardUuid);
+            return !(storageFile.exists() && storageFile.length() == backupFile.length());
```

Comment:
I think is easier to read if you invert the conditions so the `!` is not required.